Mathlib/Analysis/Complex/ValueDistribution/CountingFunction.lean

```diff
@@ -0,0 +1,216 @@
+measure of the number of times the function `f` takes a given value `a` within the disk `∣z∣ ≤ r`,
+counting multiplicities.  In the special case where `a = ⊤`, it counts the poles of `f`.
+-/
+noncomputable def logCounting :
+    ℝ → ℝ := by
+  by_cases h : a = ⊤
+  · exact (divisor f univ)⁻.logCounting
+  · exact (divisor (fun z ↦ f z - a.untop₀) univ)⁺.logCounting
+
+/--
+For finite values `a₀`, the logarithmic counting function `logCounting f a₀` is the counting
+function associated with the zero-divisor of the meromorphic function `f - a₀`.
+-/
+lemma logCounting_finite :
+    logCounting f a₀ = (divisor (fun z ↦ f z - a₀) univ)⁺.logCounting := by
+  simp [logCounting]
+
+/--
+For finite values `a₀`, the logarithmic counting function `logCounting f a₀` is equals the
+logarithmic counting function for the value zero of the shifted function `f - a₀`.
+-/
+lemma logCounting_finite_eq_logCounting_zero_of_shifted :
+    logCounting f a₀ = logCounting (f - fun _ ↦ a₀) 0 := by
+  simp [logCounting]
+
+/--
+The logarithmic counting function `logCounting f 0` is the counting function associated with the
+zero-divisor of `f`.
+-/
+lemma logCounting_zero :
+    logCounting f 0 = (divisor f univ)⁺.logCounting := by
+  simp [logCounting]
+
+/--
+The logarithmic counting function `logCounting f Set.univ` is the counting function associated with
+the pole-divisor of `f`.
+-/
+lemma logCounting_top :
+    logCounting f ⊤ = (divisor f univ)⁻.logCounting := by
+  simp [logCounting]
+
+/--
+Evaluation of the logarithmic counting function at zero yields zero.
+-/
+lemma logCounting_eval_zero :
+    logCounting f a 0 = 0 := by
+  by_cases h : a = ⊤ <;> simp [logCounting, h]
+
+/--
+The counting function associated with the divisor of `f` is the difference between `logCounting f 0`
+and `logCounting f ⊤`.
+-/
+theorem log_counting_zero_sub_logCounting_top {f : 𝕜 → E} :
+    (divisor f univ).logCounting = logCounting f 0 - logCounting f ⊤ := by
+  rw [← posPart_sub_negPart (divisor f univ), logCounting_zero, logCounting_top, map_sub]
+
+/-!
+## Elementary Properties of the Counting Function
+-/
+
+/--
+Relation between the logarithmic counting function of `f` and of `f⁻¹`.
+-/
+theorem logCounting_inv [CompleteSpace 𝕜] {f : 𝕜 → 𝕜} :
+    logCounting f 0 = logCounting f⁻¹ ⊤ := by
+  simp [logCounting_zero, logCounting_top]
+
+/--
+Adding an analytic function does not change the counting function counting poles.
+-/
+theorem logCounting_add_analyticOn_right (hf : MeromorphicOn f univ) (hg : AnalyticOn 𝕜 g univ) :
+    logCounting (f + g) ⊤ = logCounting f ⊤ := by
+  simp only [logCounting, ↓reduceDIte]
+  rw [hf.negPart_divisor_add_of_analyticNhdOn_right (isOpen_univ.analyticOn_iff_analyticOnNhd.1 hg)]
+
+/--
+Special case of `VD.logCounting_add_analyticOn_right`: Adding a constant does not change the
+counting function counting poles.
+-/
+theorem logCounting_add_const_right (hf : MeromorphicOn f univ) :
```

Comment:
```suggestion
theorem logCounting_add_const (hf : MeromorphicOn f univ) :
```
no need for `right`, as the other one would be called `const_add`. Same thing in next lemma.

Reply:
Fixed.